ibm_cloud_sdk_core/cp4d_token_manager.py

```diff
@@ -0,0 +1,76 @@
+# coding: utf-8
+
+# Copyright 2019 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .jwt_token_manager import JWTTokenManager
+
+
+class CP4DTokenManager(JWTTokenManager):
+    TOKEN_NAME = 'accessToken'
+
+    def __init__(self,
+                 username,
+                 password,
+                 url,
+                 disable_ssl_verification=False,
+                 headers=None,
+                 proxies=None):
+        """
+        :attr str username: The username
+        :attr str password: The password
+        :attr str url: The url for authentication
+        :attr bool disable_ssl_verification: enables/ disabled ssl verification
+        :attr dict headers: user-defined headers
+        :attr dict proxies: user-defined proxies
+        """
+        self.username = username
+        self.password = password
+        url = url + '/v1/preauth/validateAuth' if url else None
```

Comment:
You should check if `url` already ends with the string `'/v1/preauth/validateAuth'` and only add that if it's not there.

Reply:
Good suggestion!